### DISCUSSION TITLE
Response Headers

### DISCUSSION THREAD

**Discussion by hjanuschka** [overblog/GraphQLBundle#1002] [General]
| Q                | A
| ---------------- | -----
| Bug report?      |  no
| Feature request? | no
| BC Break report? |  no
| RFC?             | no
| Version/Branch   |  master

we'd like to add custom response headers to the graphql requests.
like - from the resolver, depending on the query/model

is there any way, or already a sample, on how to add headers to the $response, from within a QueryInterface?

**Comment by mcg-web** [comment]
Using the [kernel.response event](https://symfony.com/doc/current/reference/events.html#kernel-response) does not answer to your needs?

**Reply by hjanuschka** [reply]
i could add headers there. BUT i need different ones depending on queries/resolvers involved.

my resolver implements ` QueryInterface, AliasedInterface `  - and on resolve i kind of like to modify the response object.
not sure, where to stick the information, to later on use it in kernel.response - you have any idea?


it would also work if i could do that on Query level instead of resolver level - but same problem.
